### PR TITLE
Re-add windows ssh key

### PR DESCRIPTION
**Pull request by bmc-msft**
Adds a scaleset specific setup script, which allows us to save the scaleset based SSH keys into the VM on setup.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/extension.py [hunk 1]
```diff
@@ -94,9 +84,24 @@ def dependency_extension(region: Region, vm_os: OS) -> Optional[Extension]:
...
+def build_scaleset_script(pool: Pool, scaleset: Scaleset) -> str:
+    commands = []
+    extension = "ps1" if pool.os == OS.windows else "sh"
+    filename = f"{scaleset.scaleset_id}/scaleset-setup.{extension}"
+    sep = "\r\n" if pool.os == OS.windows else "\n"
+
+    if pool.os == OS.windows and scaleset.auth is not None:
+        ssh_key = scaleset.auth.public_key.strip()
+        ssh_path = "$env:ProgramData/ssh/administrators_authorized_keys"
+        commands += ['Set-Content -Path %s -Value "%s"' % (ssh_path, ssh_key)]
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/extension.py:96]: Is there a reason we don't use f-strings here, though we do above?

File: src/api-service/__app__/onefuzzlib/extension.py [hunk 2]
```diff
@@ -240,10 +247,10 @@ def agent_config(
     raise NotImplementedError("unsupported OS: %s" % vm_os)
 
 
-def fuzz_extensions(region: Region, vm_os: OS, pool_name: str) -> List[Extension]:
-    urls = [build_pool_config(pool_name)]
-    fuzz_extension = agent_config(region, vm_os, AgentMode.fuzz, urls=urls)
-    extensions = generic_extensions(region, vm_os)
+def fuzz_extensions(pool: Pool, scaleset: Scaleset) -> List[Extension]:
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/extension.py:250]: Nice removal of redundancy, here.

File: src/runtime-tools/linux/setup.sh [hunk 3]
```diff
@@ -44,6 +45,10 @@ fi
 if [ -f /onefuzz/downloaded/repro-stdout.sh ]; then
     mv /onefuzz/downloaded/repro-stdout.sh /onefuzz/bin/
 fi
+if [ -f /onefuzz/downloaded/scaleset-setup.sh ]; then
+    mv /onefuzz/downloaded/scaleset-setup.sh /onefuzz/bin
```
Comment (ranweiler) [inline src/runtime-tools/linux/setup.sh:49]: Consider adding a trailing slash to the dst.

**Review by ranweiler — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/extension.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/extension.py]: Habit.  Updated.

**Pull request merged**